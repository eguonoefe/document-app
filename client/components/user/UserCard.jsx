import React from 'react';
import PropTypes from 'prop-types';
import parseDate from '../../utilities/parseDate';

/**
* User Card for viewing a single user
*
* @param {Object} props { firstName, lastName,
* email, Role, id, createdAt, onClick }
* @returns {String} HTML markup for the User Card
*/
const UserCard = (props) => {
  const { firstName, lastName, email, Role, id, createdAt, onClick } = props;
  return (
    <div>
      <a
        onClick={onClick}
        name={id}
        className="truncate"
        href="#!">{firstName} {lastName}</a>
      <p>{email}</p>
      {(Role.title === 'regularuser') && <p>Regular User</p>}
      {(Role.title === 'contributor') && <p>Contributor</p>}
      <p>Joined at: {parseDate(createdAt)}</p>
      <div className="divider" />
    </div>
  );
};

// Set Default Props
UserCard.defaultProps = {
  title: '',
  content: '',
  id: ''
};
// Set Props
UserCard.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  email: PropTypes.string,
  Role: PropTypes.object,
  createdAt: PropTypes.string,
  onClick: PropTypes.func,
  id: PropTypes.number
};

export default UserCard;
