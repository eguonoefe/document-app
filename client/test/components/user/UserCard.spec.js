/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import UserCard from '../../../components/user/UserCard';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/user/UserCard');

describe('UserCard Component', () => {
  test('should match the user card snapshot', () => {
    const onClick = jest.fn();
    const Role = {
      title: '',
      description: ''
    };
    const component = shallow(
      <UserCard
        firstName={'Eguono'}
        id={3}
        lastName={'Efe'}
        email={'efe@gmail.com'}
        onClick={onClick}
        Role={Role}
        createdAt={'today'} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
