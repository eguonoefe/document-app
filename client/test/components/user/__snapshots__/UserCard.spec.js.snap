// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Component should match the user card snapshot 1`] = `
<div>
  <a
    className="truncate"
    href="#!"
    name={3}
    onClick={[MockFunction]}
  >
    Eguono
     
    Efe
  </a>
  <p>
    efe@gmail.com
  </p>
  <p>
    Joined at: 
    Invalid Date
  </p>
  <div
    className="divider"
  />
</div>
`;
